dask/dataframe/optimize.py

```diff
@@ -63,11 +63,19 @@ def optimize_dataframe_getitem(dsk, keys):
 
     layers = dsk.layers.copy()
     dependencies = dsk.dependencies.copy()
-
+    column_projection = True
     for k in dataframe_blockwise:
```

Comment:
Should this go inside the `for k in dataframe_blockwise:` loop below? It looks like before we were handling column projection for each layer in `dataframe_blockwise` independently and now we're linking them together with `column_projection`. Just wanted to make sure this was intentional

Reply:
yup - my mistake!